thrust/thrust/system/cuda/detail/execution_policy.h

```diff
@@ -40,11 +40,14 @@
 #include <thrust/system/cuda/config.h>
 
 #include <thrust/detail/allocator_aware_execution_policy.h>
-#include <thrust/detail/dependencies_aware_execution_policy.h>
 #include <thrust/detail/execution_policy.h>
 #include <thrust/iterator/detail/any_system_tag.h>
 #include <thrust/version.h>
 
+#if !_CCCL_COMPILER(NVRTC)
+#  include <thrust/detail/dependencies_aware_execution_policy.h>
```

Comment:
Is this intended to be backported?

Otherwise we could just drop this universally

Reply:
This PR enables us to use thrust iterators instead of the deprecated CUB iterators with a 100 buggy warning suppressions on the 2.8.x. branch. It does not need to be backported, but if we see too many issues with the warning suppressions, this PR + a replacement of CUB by thrust iterators gives us a way out.